src/main/java/org/jenkinsci/plugins/workflow/actions/ExecutorTaskInfoAction.java

```diff
@@ -0,0 +1,88 @@
+package org.jenkinsci.plugins.workflow.actions;
+
+import hudson.model.InvisibleAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+/**
+ * Records information for a {@code node} block.
+ */
+public class ExecutorTaskInfoAction extends InvisibleAction implements PersistentAction {
+    private static final long serialVersionUID = 1;
+
+    private String whyBlocked;
+    private boolean launched;
+    // Initialized at -1 for "not started yet".
+    private long whenStartedOrCancelled = -1L;
```

Comment:
:bug:  Can't we obtain this by looking at the flowgraph, specifically the timings of the nested blocks for the ExecutorStep/WorkspaceStep?   The TimingAction should give us when we started IIRC.